src/test/kotlin/uk/gov/dluhc/printapi/mapper/PrintDetailsMapperTest.kt

```diff
@@ -0,0 +1,135 @@
+package uk.gov.dluhc.printapi.mapper
+
+import org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.any
+import org.mockito.kotlin.given
+import org.mockito.kotlin.verify
+import uk.gov.dluhc.printapi.database.entity.Address
+import uk.gov.dluhc.printapi.database.entity.CertificateDelivery
+import uk.gov.dluhc.printapi.database.entity.DeliveryClass
+import uk.gov.dluhc.printapi.database.entity.ElectoralRegistrationOffice
+import uk.gov.dluhc.printapi.database.entity.PrintDetails
+import uk.gov.dluhc.printapi.messaging.models.CertificateLanguage
+import uk.gov.dluhc.printapi.testsupport.testdata.dto.buildEroManagementApiEroDto
+import uk.gov.dluhc.printapi.testsupport.testdata.getAMongoDbId
+import uk.gov.dluhc.printapi.testsupport.testdata.model.buildSendApplicationToPrintMessage
+import java.time.LocalDate
+import java.util.UUID
+import uk.gov.dluhc.printapi.database.entity.CertificateLanguage as CertificateLanguageEntity
+import uk.gov.dluhc.printapi.database.entity.SourceType as SourceTypeEntity
+import uk.gov.dluhc.printapi.messaging.models.CertificateLanguage as CertificateLanguageModel
+import uk.gov.dluhc.printapi.messaging.models.SourceType as SourceTypeModel
+
+@ExtendWith(MockitoExtension::class)
+class PrintDetailsMapperTest {
+    @InjectMocks
+    private lateinit var mapper: PrintDetailsMapperImpl
+
+    @Mock
+    private lateinit var sourceTypeMapper: SourceTypeMapper
+
+    @ParameterizedTest
+    @CsvSource(value = ["EN, EN", "CY, CY"])
+    fun `should map send application to print message to print details`(
+        certificateLanguageModel: CertificateLanguageModel,
+        certificateLanguageEntity: CertificateLanguageEntity
+    ) {
+        // Given
+        val ero = buildEroManagementApiEroDto()
+        val localAuthority = ero.localAuthorities[0]
+        val message = buildSendApplicationToPrintMessage(certificateLanguage = certificateLanguageModel)
+        given(sourceTypeMapper.toSourceTypeEntity(any())).willReturn(SourceTypeEntity.VOTER_CARD)
+
+        val expected = with(message) {
+            PrintDetails(
+                id = UUID.randomUUID(),
+                requestId = getAMongoDbId(),
```

Comment:
I'd like to abstract us away from the underlying technology if we could please?
The only reason we are using a mongo ID is that its convenient and suits the business need. As you know I'd rather we weren't using the mongo library at all, because we're not using mongo! ¯\_(ツ)_/¯ 

Presumably the method `getAMongoId` in a test data builder somewhere? Could you rename it please to something like `getARequestId` - because from a business perspective thats what its doing.
Thanks 🙏